### PR TITLE
fix(docker): Correct dev commands for Nextjs, build before start in prod

### PR DESCRIPTION
**Pull request by zachblume**
Saw the repo and wanted to poke around a bit. Didn't manage to get things fully set up even in dev but made some progress and took some friction notes along the way.

This PR patches the docker compose in dev and prod to include correct nextjs build and dev-start commands, and avoid typesafety errors bubbling on the frontend for a clearer error if auth isn't fully wired up correctly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,22 +104,23 @@ git clone https://github.com/turntable-so/turntable.git
...
-
-```bash
-echo "ENCRYPTION_KEY=\"`openssl rand -base64 32`\"" >> .env
-```
 
 3. Start the app
 
 ```bash
-docker compose up --env-file .env
+docker compose up
```
Comment (zachblume) [inline README.md:123]: `--env-file` is a command for `docker run` but `docker compose` doesn't accept it

File: backend/api/settings.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 # See https://docs.djangoproject.com/en/5.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = "django-insecure-i7pd5iwtuo*0h9je%(n1!u8srlbka$^do)c(#y88h9grhilq3@"
+SECRET_KEY = "django-insecure-i7pd5iwtuo*0h9je%(n1!u8srlbka$^do)c(#y88h9grhilq3@" # oooo?
```
Comment (zachblume) [inline backend/api/settings.py:27]: Live secret?

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -49,6 +49,7 @@ services:
       DEV: "true"
     volumes:
       - ./frontend:/code
+    command: ["pnpm", "dev"]
```
Comment (zachblume) [inline docker-compose.dev.yml:52]: Override the production `start` command with `dev` for next hot reloading

File: docker-compose.yml [hunk 4]
```diff
@@ -96,7 +96,7 @@ services:
     build:
       context: frontend
       dockerfile: Dockerfile
-    command: ["pnpm", "start"]
+    command: ["pnpm", "build", "&&", "pnpm", "start"]
```
Comment (zachblume) [inline docker-compose.yml:99]: Need to build before starting nextjs

File: frontend/src/components/auth/registration-form.tsx [hunk 5]
```diff
@@ -80,13 +80,16 @@ const RegistrationForm = ({invitationCode = ''} : any) => {
             .catch((err) => {
                 console.log("ERROR")
                 console.log(err)
-                setIsLoading(false)
-                if (err.json.email) {
+                if (err?.json?.email) {
```
Comment (zachblume) [inline frontend/src/components/auth/registration-form.tsx:83]: Unsafe accessor just bubbles an error from the catch() if fetch fails, this gives a clearer error in that case

File: README.md (hunk 1 above)
Comment (ianrtracey) [reply README.md:123]: Hi @zachblume, which version of docker compose are you using? We use this command internally and found it in the docs here:
https://docs.docker.com/compose/environment-variables/variable-interpolation/

File: backend/api/settings.py (hunk 2 above)
Comment (ianrtracey) [reply backend/api/settings.py:27]: This is non-live for users of our open source product

File: frontend/src/components/auth/registration-form.tsx (hunk 5 above)
Comment (ianrtracey) [reply frontend/src/components/auth/registration-form.tsx:83]: looks great!

File: README.md (hunk 1 above)
Comment (zachblume) [reply README.md:123]: Ah, looks like this is why:
```
 ~/dev/turntable | zachblume/fi..d-prod-build  docker --version                                                                                                                           ok | base py 

Docker version 26.1.3, build b72abbb
 ~/dev/turntable | zachblume/fi..d-prod-build  docker compose up                                                                                                                          ok | base py 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
[+] Running 33/5
 ⠸ postgres [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿] 151.8MB / 152.6MB Pulling                                                                                                                                          10.4s 
 ⠸ setup-hatchet [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣶⡀⣿⣄⠀⠀⠀⠀⠀⠀] Pulling                                                                                                                                                    10.4s 
 ⠸ hatchet-lite Pulling                                                                                                                                                                           10.4s 
 ✔ minio Pulled                                                                                                                                                                                    8.9s 
 ⠸ createbuckets [⠀⠀⠀⠀⠀] Pulling                                                                                                                                                                  10.4s 
canceled
 ~/dev/turntable | zachblume/fi..d-prod-build  docker compose up --env-file .env                                                                                                   INT | 11s | base py 
unknown flag: --env-file
 ~/dev/turntable | zachblume/fi..d-prod-build  docker compose --env-file .env up                                                                                                      16 err | base py 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "OPENAI_API_KEY" variable is not set. Defaulting to a blank string. 
[+] Running 6/4
 ⠇ setup-hatchet Pulling                                                                                                                                                                           1.8s 
 ⠇ hatchet-lite [⣿⣦⣿⣿⣤⣿⣿⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀] Pulling                                                                                                                                                      1.8s 
 ⠇ createbuckets [⣿⠀⠀⠀⠀⠀] Pulling                                                                                                                                                                  1.8s 
 ⠇ postgres [⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀] Pulling                                                                                                                                                             1.8s 
canceled
 ~/dev/turntable | zachblume/fi..d-prod-build                                                                                                                                            INT | base py 
 ~/dev/turntable | zachblume/fi..d-prod-build                                                                                                                                            INT | base py 
```

You must specify the env file flag before the `up` command

Comment (zachblume) [reply README.md]: I added the flag back into this PR in the correct position

File: docker-compose.dev.yml [hunk 6]
```diff
@@ -48,7 +48,8 @@ services:
     environment:
       DEV: "true"
     volumes:
-      - ./frontend:/code
+      - ./frontend:/
```
Comment (zachblume) [inline docker-compose.dev.yml:51]: Found one more issue here: the volume was being mounted to /code but the entrypoint of frontend is left as root. So this actually mounts the frontend folder as root of this container for dev

**Review by ianrtracey — APPROVED** [review]

**Pull request closed without merging**